README.md

```diff
@@ -56,10 +71,186 @@ These three files can be downloaded directly from `inputs` and `dataset` directo
+             pretrained_vocab_path="./models/finetuned_test001/finetuned_test001.vocab",
+             inference_mode="vect",
+             scenario="candidates/test")
+```
+
+* [Assembling queries vector representations](#candidate-ranker-and-assembling-vector-representations):
+
+```python
+from DeezyMatch import combine_vecs
+
+# combine vectors stored in queries/test and save them in combined/queries_test
+combine_vecs(rnn_passes=['fwd', 'bwd'], 
+             input_scenario='queries/test', 
+             output_scenario='combined/queries_test', 
+             print_every=10)
+```
+
+* [Assembling candidates vector representations](#candidate-ranker-and-assembling-vector-representations):
+
+```python
+from DeezyMatch import combine_vecs
+
+# combine vectors stored in candidates/test and save them in combined/candidates_test
+combine_vecs(rnn_passes=['fwd', 'bwd'], 
+             input_scenario='candidates/test', 
+             output_scenario='combined/candidates_test', 
+             print_every=10)
+```
+
+* [Candidate ranker](#candidate-ranker-and-assembling-vector-representations):
+
+```python
+
+from DeezyMatch import candidate_ranker
+
+# Select candidates based on L2-norm distance (aka faiss distance):
+# find candidates from candidate_scenario 
+# for queries specified in query_scenario
+candidates_pd = \
+    candidate_ranker(query_scenario="./combined/queries_test",
+                     candidate_scenario="./combined/candidates_test", 
+                     ranking_metric="faiss", 
+                     selection_threshold=5., 
+                     num_candidates=2, 
+                     search_size=2, 
+                     output_path="ranker_results/test_candidates_deezymatch", 
+                     pretrained_model_path="./models/finetuned_test001/finetuned_test001.model", 
+                     pretrained_vocab_path="./models/finetuned_test001/finetuned_test001.vocab", 
+                     number_test_rows=20)
+```
+
+* [Candidate ranking on-the-fly](#candidate-ranking-on-the-fly):
+
+```python
+from DeezyMatch import candidate_ranker
+
+# Ranking on-the-fly
+# find candidates from candidate_scenario 
+# for queries specified by the `query` argument
+candidates_pd = \
+    candidate_ranker(candidate_scenario="./combined/candidates_test",
+                     query=["sinbotin", "DeezyMatch", "kruty"],
+                     ranking_metric="faiss", 
+                     selection_threshold=5., 
+                     num_candidates=1, 
+                     search_size=100, 
+                     output_path="ranker_results/test_candidates_deezymatch_on_the_fly", 
+                     pretrained_model_path="./models/finetuned_test001/finetuned_test001.model", 
+                     pretrained_vocab_path="./models/finetuned_test001/finetuned_test001.vocab", 
+                     number_test_rows=20)
+```
+
+To avoid entering all the inputs every time, candidate ranker can be called via:
```

Comment:
Maybe I'll change this a bit:
"The candidate ranker can be initialised, to be used multiple times, by running:" 

Does it keep the meaning?

Reply:
Thanks, changed.